class.jetpack.php

```diff
@@ -2884,8 +2884,63 @@ public static function plugin_activation( $network_wide ) {
 		// For firing one-off events (notices) immediately after activation
 		set_transient( 'activated_jetpack', true, .1 * MINUTE_IN_SECONDS );
 
+		update_option( 'jetpack_activation_source', self::get_activation_source( wp_get_referer() ) );
+
 		Jetpack::plugin_initialize();
 	}
+
+	public static function get_activation_source( $referer_url ) {
+
+		if ( defined( 'WP_CLI' ) && WP_CLI ) {
+			return array( 'wp-cli', null );
+		}
+
+		$referer = parse_url( $referer_url );
+		
+		if ( ! is_array( $referer ) ) {
+			return array( 'unknown', null );
+		}
+
+		$plugins_path = parse_url( admin_url( 'plugins.php' ), PHP_URL_PATH );
+		$plugins_install_path = parse_url( admin_url( 'plugin-install.php' ), PHP_URL_PATH );// /wp-admin/plugin-install.php
+		
+		if ( isset( $referer['query'] ) ) {
+			parse_str( $referer['query'], $query_parts );
+		} else {
+			$query_parts = array();
+		}
+
+		$source_type = false;
+		$source_query = null;
```

Comment:
I could see some random old php version actually storing the string value `null` ... would it make sense to just start with an empty string?

Reply:
Which PHP version do you think would do this? I could add a test.

Reply:
PHP 5.2 never ceases to amaze me